pkg/apiserver/slowquery/model.go

```diff
@@ -0,0 +1,129 @@
+	BackoffTypes string `gorm:"column:Backoff_types" json:"backoff_types"`
+
+	// Connection
+	User string `gorm:"column:User" json:"user"`
+	Host string `gorm:"column:Host" json:"host"`
+
+	// Time
+	ProcessTime            float64 `gorm:"column:Process_time" json:"process_time"`
+	WaitTime               float64 `gorm:"column:Wait_time" json:"wait_time"`
+	BackoffTime            float64 `gorm:"column:Backoff_time" json:"backoff_time"`
+	GetCommitTSTime        float64 `gorm:"column:Get_commit_ts_time" json:"get_commit_ts_time"`
+	LocalLatchWaitTime     float64 `gorm:"column:Local_latch_wait_time" json:"local_latch_wait_time"`
+	ResolveLockTime        float64 `gorm:"column:Resolve_lock_time" json:"resolve_lock_time"`
+	PrewriteTime           float64 `gorm:"column:Prewrite_time" json:"prewrite_time"`
+	WaitPreWriteBinlogTime float64 `gorm:"column:Wait_prewrite_binlog_time" json:"wait_prewrite_binlog_time"`
+	CommitTime             float64 `gorm:"column:Commit_time" json:"commit_time"`
+	CommitBackoffTime      float64 `gorm:"column:Commit_backoff_time" json:"commit_backoff_time"`
+	CopProcAvg             float64 `gorm:"column:Cop_proc_avg" json:"cop_proc_avg"`
+	CopProcP90             float64 `gorm:"column:Cop_proc_p90" json:"cop_proc_p90"`
+	CopProcMax             float64 `gorm:"column:Cop_proc_max" json:"cop_proc_max"`
+	CopWaitAvg             float64 `gorm:"column:Cop_wait_avg" json:"cop_wait_avg"`
+	CopWaitP90             float64 `gorm:"column:Cop_wait_p90" json:"cop_wait_p90"`
+	CopWaitMax             float64 `gorm:"column:Cop_wait_max" json:"cop_wait_max"`
+
+	// Transaction
+	WriteKeys      int `gorm:"column:Write_keys" json:"write_keys"`
+	WriteSize      int `gorm:"column:Write_size" json:"write_size"`
+	PrewriteRegion int `gorm:"column:Prewrite_region" json:"prewrite_region"`
+	TxnRetry       int `gorm:"column:Txn_retry" json:"txn_retry"`
+
+	// Coprocessor
+	RequestCount uint   `gorm:"column:Request_count" json:"request_count"`
+	ProcessKeys  uint   `gorm:"column:Process_keys" json:"process_keys"`
+	TotalKeys    uint   `gorm:"column:Total_keys" json:"total_keys"`
+	CopProcAddr  string `gorm:"column:Cop_proc_addr" json:"cop_proc_addr"`
+	CopWaitAddr  string `gorm:"column:Cop_wait_addr" json:"cop_wait_addr"`
+
+	// RocksDB
+	RocksdbDeleteSkippedCount uint `gorm:"column:Rocksdb_delete_skipped_count" json:"rocksdb_delete_skipped_count"`
+	RocksdbKeySkippedCount    uint `gorm:"column:Rocksdb_key_skipped_count" json:"rocksdb_key_skipped_count"`
+	RocksdbBlockCacheHitCount uint `gorm:"column:Rocksdb_block_cache_hit_count" json:"rocksdb_block_cache_hit_count"`
+	RocksdbBlockReadCount     uint `gorm:"column:Rocksdb_block_read_count" json:"rocksdb_block_read_count"`
+	RocksdbBlockReadByte      uint `gorm:"column:Rocksdb_block_read_byte" json:"rocksdb_block_read_byte"`
+}
+
+type Field struct {
+	ColumnName string
+	JSONName   string
+	Projection string
+}
+
+func getFieldsAndTags() (slowQueryFields []Field) {
```

Comment:
can we cache it since it's an invariant and can be determined at compile time 😀

Reply:
Yes, I thought about generating the code here before. Collected in todo.